### ISSUE TITLE
Service-to-Service invocation for "External" services

### ISSUE THREAD

**Issue by trondhindenes** [dapr/dapr#3545]
## In what area(s)?

/area runtime

> /area operator

> /area placement

> /area docs

> /area test-and-release
## Describe the proposal
Dapr support service-to-service invocation between "dapr-ized" services. It would be interesting to expand this to services outside dapr. My concrete use-case is invoking requests against an AWS Api gateway, or other api gateways that require some kind of specialized authentication logic. For AWS Api Gateway, the request data is signed along with the aws credential and use by API Gateway to verify the caller's identity. This is non-trivial to in code, and would be awesome to be able to hand off to the dapr sidecar. 

In order for this to work, the external service would have to be registered in dapr with a name and a type, and the type would govern what kind of "preprocessing" dapr would do on the outgoing request (such as signing the request).

I think this would open up some interesting opportunities. I'd be happy to try and help develop the spec and implement a proof-of-concept for aws api gateway.

**Comment by cvictory** [comment]
Good Idea !  Maybe the service invocation should implements as another component.
